exla/c_src/exla/mlir/builder.cc

```diff
@@ -105,11 +116,71 @@ MLIRModule::MLIRModule() {
 
 mlir::Type TypeIntToMLIRType(mlir::OpBuilder* builder, int type_int) {
   switch (type_int) {
+    case 2:
+      return builder->getIntegerType(8);
+    case 3:
+      return builder->getIntegerType(16);
+    case 4:
+      return builder->getIntegerType(32);
+    case 5:
+      return builder->getIntegerType(64);
+    case 6:
+      return builder->getIntegerType(8, false);
+    case 7:
+      return builder->getIntegerType(16, false);
+    case 8:
+      return builder->getIntegerType(32, false);
     case 9:
+      return builder->getIntegerType(64, false);
+    case 10:
+      return builder->getF16Type();
+    case 11:
       return builder->getF32Type();
+    case 12:
+      return builder->getF64Type();
+    case 16:
+      return builder->getBF16Type();
   }
 }
 
+xla::PrimitiveType MLIRTypeToPrimitiveType(mlir::Type type) {
```

Comment:
What do you think is the benefit of still using primitive types?

Reply:
I didn't think much about benefits, more about having to use them for getting the shape information

Reply:
There might be a way of getting around them

Reply:
Yeah, so this is something we'll have to deal with some time in the future:

I tried making it so that we could return a ref to the ShapedTypeComponents object as the shape in case of the MLIR implementation, but I got segfaulted into oblivion. So I'm gonna leave this MLIRTypeToPrimitiveType conversion for usage in the building of the xla::Shape in mlir_get_shape

Reply:
This refactor will probably clean up some other things like the TypeIntToMLIRType function too, but I think we can deal with this later